app/components-react/windows/NameFolder.tsx

```diff
@@ -0,0 +1,80 @@
+import { ModalLayout } from '../shared/ModalLayout';
+import { $t } from '../../services/i18n';
+import React, { useState } from 'react';
+import { Services } from '../service-provider';
+import { useOnce } from '../hooks';
+import { assertIsDefined } from '../../util/properties-type-guards';
+
+interface IWindowOptions {
+  renameId?: string;
+  itemsToGroup?: string[];
+  parentId?: string;
+  sceneId: string;
+}
+
+/**
+ * Modal for creating or re-naming a folder
+ */
+export default function NameFolder() {
+  // inject services
+  const { ScenesService, EditorCommandsService, WindowsService } = Services;
+
+  // define stateful variables and setters
+  const [name, setName] = useState('');
+  const [error, setError] = useState('');
```

Comment:
is it normal to use `useState` multiple times like this? feels a little like it could be collapsed into an object

Reply:
You may collapse it into a single object if you need. For me, it's was convenient to have separate setters here. So I can use more readable `setError(myError)` instead `setState({ error: myError, ...state })`

Reply:
Yeah in my very limited knowledge of hooks, each call to `setState` gives you a single state variable.  You can choose to make that variable a complex object containing all your state, or you can use lots of separate variable.  The react docs don't seem to indicate one approach is better than others, so seems up to the dev and what is more clear.  I'm sure we will end up heading in one direction or another as we get used to the patterns.